File: Inventory_Backend/src/main/java/com/Inventory/Inventory_Backend/ewaybill/service/EWayBillService.java
package com.Inventory.Inventory_Backend.ewaybill.service;

import com.Inventory.Inventory_Backend.ewaybill.dto.*;
import org.springframework.data.domain.Page;

import java.util.List;

public interface EWayBillService {

    EWayBillResponse createEWayBill(Long businessId, EWayBillCreateRequest request);

    EWayBillResponse getEWayBill(Long businessId, Long id);

    List<EWayBillResponse> getAllEWayBills(Long businessId);

    EWayBillResponse updateEWayBill(Long businessId, Long id, EWayBillUpdateRequest request);

    void deleteEWayBill(Long businessId, Long id);

    EWayBillResponse updateVehicle(Long businessId, Long id, EWayBillVehicleUpdateRequest request);

    EWayBillResponse extendValidity(Long businessId, Long id, EWayBillExtendValidityRequest request);

    void cancelEWayBill(Long businessId, Long id);

    Page<EWayBillResponse> getAllBills(Long businessId, int page, int size);

}

File: frontend/src/services/ewayBillService.js
import api from './api';

export const createEWayBill = (data) => 
    api.post(`/eway-bills`, data);

export const getAllEWayBills = (page = 0, size = 50) =>
    api.get(`/eway-bills?page=${page}&size=${size}`);

export const getEWayBillById = (id) => 
    api.get(`/eway-bills/${id}`);

export const updateEWayBill = (id, data) =>
    api.put(`/eway-bills/${id}`, data);

export const deleteEWayBill = (id) =>
    api.delete(`/eway-bills/${id}`);

export const updateVehicle = (id, data) =>
    api.patch(`/eway-bills/${id}/vehicle`, data);

export const extendValidity = (id, data) =>
    api.patch(`/eway-bills/${id}/extend-validity`, data);

export const cancelEWayBill = (id) =>
    api.patch(`/eway-bills/${id}/cancel`);